api/organisations/views.py

```diff
@@ -274,10 +274,12 @@ def chargebee_webhook(request: Request) -> Response:
     """
     event_type = request.data.get("event_type")
 
-    if event_type == "payment_failed":
+    if event_type == webhook_event_types.PAYMENT_FAILED:
         return webhook_handlers.payment_failed(request)
-    if event_type == "payment_succeeded":
+    if event_type == webhook_event_types.PAYMENT_SUCCEEDED:
         return webhook_handlers.payment_succeeded(request)
+    if event_type in webhook_event_types.CACHE_REBUILD_TYPES:
+        return webhook_handlers.cache_rebuild_event(request)
```

Comment:
I really wish there was a better way to do this instead of just a (probably eventually) large conditional statement. One other option would be to use a `match` / `case` but it seems there's no easy way to make it play nice with checking if an item is in a list. 

From a personal preference point of view, unless there's a reason not to, I'd prefer this to be written using `if` / `elif` just as it makes it slightly more obvious that it can only match one condition (although I guess that's the case given the fact that we're returning in each case anyway).

Reply:
I've been raised as an early exit from an `if` means that the next exit starts as an `if` not an `elif`. I can't remember which linter enforces this, but there was one in the past. 

As for this turning into a large conditional statement, provided once we refactor this block of code I hope that every statement looks like this:

```python
if conditional1:
    return func_1()
if conditional2:
    return func_2()
if conditional3:
    return func_3()

return None
```

If the if checks are lean then the person reviewing the code can quickly understand what happens and in which order of preference. It's when `if` statements get subnested and variables get assigned from one sub `if` statement to another that it gets hard to reason about. At the end of the day, we receive all their webhooks through a single view so we have to filter them somehow. If we try to create a layer of indirection to, say, register functions based on the `event_type` of the post body then it may look more pleasing, but it's overhead that's doing the same thing. I think we should keep it simple.

Reply:
Ok, agreed, thanks for the comprehensive response!